### ISSUE TITLE
[csolution] deprecate support for -p option and mandate csolution.yml (was: Generator : -p versus -c)

### ISSUE THREAD

**Issue by fred-r** [Open-CMSIS-Pack/devtools#334]
According to help:
```
> csolution.exe run --help
CMSIS Project Manager 0.9.5 (C) 2022 ARM
...
Options:
  -s, --solution arg    Input csolution.yml file
  -p, --project arg     Input cproject.yml file
  -c, --context arg     Input context name <cproject>[.<build-type>][+<target-type>]
```

But:
```
> csolution.exe run -g STM32Cube_BasicTrace_CodeGen -c gpio-toggle.AC6+B-U585I-IOT02A -s gpio-toggle.csolution.yml
```
works fine.

while:
```
> csolution.exe run -g STM32Cube_BasicTrace_CodeGen -p gpio-toggle.cproject.yml -s gpio-toggle.csolution.yml
error csolution: context was not specified
```

and:
```
> csolution.exe run -g STM32Cube_BasicTrace_CodeGen -c gpio-toggle.AC6+B-U585I-IOT02A -p gpio-toggle.cproject.yml -s gpio-toggle.csolution.yml
```

works too.

I agree that only the context knows the build-type and this can change the components to be considered.

But why having -p then ?
When can we use it ? What for ?

@jeromecoutant

**Comment by jkrech** [comment]
@fred-r we had initially thought about allowing to specify a cproject without the existence of a csolution file. In the meantime I vote for removing the -p option completely, as it otherwise requires board/device and compiler to be specified in the cproject file.
Would you have any objections to mandate a `<name>.csolution.yml` referencing the `<name>.cproject.yml` 
In this case I would not only remove `-p` but also remove `-s` as the csolution.yml file is the mandatory input file required for all commands (except `help`).

**Comment by fred-r** [comment]
Yes, to me the csolution should always be considered and -p probably makes no sense alone.

@slhultgren and @VGRSTM : what do you think ?

**Comment by VGRSTM** [comment]
@fred-r myself I would say it depends of context.
+ If commenting as ST Microelectronics contributor I would say: `csolution` is must have
+ If commenting as OpenCMSIS contributor I may say: `csolution` is an option, `cproject` is must have
+ If commenting as end user I would have feeling: perfectionism can be counter-productive

Myself I always try to take end user hat first. Having too open proposal is fine as expert but not helping at all common end user. I would prefer so to get a unique way of working. As we cannot lock expert we have to enable complexity support. Common end user has to be supported then by some UI or ...

In short I would push for : `csolution` is must have and so yes get rid of `-c` and `-p` switches.

Such said, one time more, my own ultimate mind about is such problematic is `devtools` one I'm happy about but not my OpenCMSIS primary interest. My primary interest is about format specification (i.e. *.yml, *.pdsc, *.cprj, ...)

**Comment by jkrech** [comment]
@fred-r and @VGRSTM thanks for sharing your feedback. I will bring this to the meeting to drive to conclusion. Once agreed we should open another issue for removing "unnecessary" elements from the cproject.yml schema.

**Comment by jkrech** [comment]
We close this discussion and the change request is captured here: https://github.com/Open-CMSIS-Pack/devtools/issues/348
